src/openrct2/diagnostic.c

```diff
@@ -30,6 +31,15 @@ const char * _level_strings[] = {
 	"INFO"
 };
 
+// returns the current system time in a printable format                                                                                                                                                    
+char* getTime() {
+        time_t rawTime;
+        time(&rawTime);
+        static char curTime[128];
+        strftime(curTime, sizeof(curTime), "[%FT%T]", localtime(&rawTime));
+        return curTime;
```

Comment:
Rather than returning a local (even if `static`), can you have this function receive a buffer to which it will `strftime`?